crates/optimism/rpc/src/sequencer.rs

```diff
@@ -22,17 +19,15 @@ pub struct SequencerClient {
 impl SequencerClient {
     /// Creates a new [`SequencerClient`].
     pub fn new(sequencer_endpoint: impl Into<String>) -> Self {
-        let client = Client::builder().use_rustls_tls().build().unwrap();
+        let sequencer_endpoint: String = sequencer_endpoint.into();
+        let client = Client::new_http(reqwest::Url::parse(&sequencer_endpoint).unwrap());
```

Comment:
we can keep using the reqwest::Client and use `Client::with_client` here I believe

Reply:
restored the reqwest::Client to use rust_tls and creating `RpcClient` with the help of  `alloy_transport_http::Http::with_client` and `ClientBuilder::transport` .